src/python/zquantum/vqe/utils.py

```diff
@@ -1,64 +1,59 @@
 
     Args:
-        n_mo (int): number of molecular orbitals
-        n_alpha (int): number of alpha electrons
-        n_beta (int): number of beta electrons
-        ansatz_type (str): which ansatz to use. Currently only 'singlet UCCSD' is supported.
-        transformation (str): which Fermion transformation to use
-        fermion_generator (openfermion.FermionOperator): Fermion generator for a general UCC 
-            ansatz
-        threshold (float): a threshold used in the selection of the operators composing the ansatz
-        spin_ordering (str): how spins are ordered in the qubit register, allow us
-                             to create correct HF state. 'interleaved': spins alpha and
-                             beta are alternating ; 'blocks': all spin alpha, followed  
-                             by all spin beta
-        n_qubits (int): The number of qubits. If None, it is taken to be 2 * n_mo
-        ordering (list): Ordering of the lattice sites, each pair of numbers indicates
-                                  which qubits the spin-up and spin-down lattice site corespond to.
-                                  Even indices are spin-up qubits and odd are spin-down qubits
-        layers (zquantum.core.circuit.CircuitLayers): 
-                                Object describing the layout of layers of 2-qubit gate blocks.
-                                The layers attribute is a list of list of tuples where each tuple
-                                contains the indices of the qubits where the gate block should be 
-                                aplied. The layers are applied in the order, looping over the list,
-                                until there are no more parameters.
-        has_rx_layer (bool): whether to include the layer of Rx rotations at the beginning of the 
-            ansatz.
-        elementary (bool): if True, decomposes the U1ex and U2ex gates into elementary gates. Useful
-                           if they are not defined on the device/simulator
+        fermion_generator (openfermion.FermionOperator or 
+            openfermion.InteractionOperator): fermionic generator.
+        fermion_transform (str): The name of the qubit to fermion transformation
+            to use.
 
     Returns:
-        dict: dictionary describing the ansatz.
+        zquantum.core.circuit.Circuit: Circuit corresponding to the exponentiation of the
+            transformed operator. 
     """
+    if transformation not in ["Jordan-Wigner", "Bravyi-Kitaev"]:
+        raise RuntimeError(f"Unrecognized transformation {transformation}")
+
+    # Transform generator to qubits
+    if transformation == "Jordan-Wigner":
+        transformation = jordan_wigner
+    elif transformation == "Bravyi-Kitaev":
+        transformation = bravyi_kitaev
+
+    qubit_generator = transformation(fermion_generator)
+
+    for term in qubit_generator.terms:
+        qubit_generator.terms[term] = float(qubit_generator.terms[term].imag)
```

Comment:
The code above seems to indicate this function only works for antihermitian fermionic operators, which makes sense since otherwise the exponential is not unitary. Required anti-hermiticity should maybe be specified in the docstring and there probably should be an assertion at the top of the function.

Reply:
Is there some "out of the box" way of performing such assertion?
I can write it myself, but don't want to risk any numerical issues.

Reply:
I thought that existed for `FermionicOperator` or `InteractionOperator` but it doesn't, however we have the `is_antihermitian` function of our `QubitOperator` class. As we're using openfermion's `QubitOperator` here we could just check if one of the coefficient is real with
`if not np.isclose(qubit_generator.terms[term].real, 0.): raise Something`

Reply:
Fixed in [f0be93](https://github.com/zapatacomputing/z-quantum-vqe/pull/5/commits/f0be93326563cae98c6ecb802c7959b0e8688b84)